src/charm.py

```diff
@@ -9,264 +9,93 @@
-            if ca_cert:
-                # update cacert in charm container too, for self-instrumentation
-                Path(TEMPO_CA_FILE).write_text(ca_cert)
-                subprocess.run(["update-ca-certificates", "--fresh"])
-            return True
-
-        else:
-            self.tempo.clear_certs()
-
-            # clear from charm container too
-            Path(TEMPO_CA_FILE).unlink(missing_ok=True)
-            subprocess.run(["update-ca-certificates", "--fresh"])
-            return True
-
-    def pre_update_checklist(
-        self,
-    ) -> List[str]:
-        """Check if the charm as a whole is ready, and if not, return a list of failed checks."""
-        failures = []
-        failures.extend(self.tempo.pre_update_checklist())
-
-        if not self.tempo_cluster.get_tempo_config():
-            failures.append("tempo_cluster config not ready")
-
-        return failures
-
     def _on_collect_status(self, e: CollectStatusEvent):
-        if not self._container.can_connect():
-            e.add_status(WaitingStatus("Waiting for `tempo` container"))
-        if not self.model.get_relation("tempo-cluster"):
-            e.add_status(
-                BlockedStatus("Missing tempo-cluster relation to a tempo-coordinator charm")
-            )
-        elif not self.tempo_cluster.relation:
-            e.add_status(WaitingStatus("Tempo-Cluster relation not ready"))
-        if not self.tempo_cluster.get_tempo_config():
-            e.add_status(WaitingStatus("Waiting for coordinator to publish a tempo config"))
-
-        if role := self.tempo_role:
-            e.add_status(
-                ActiveStatus(
-                    "(all roles) ready." if role is TempoRole.all else f"{role.name} ready."
-                )
-            )
-        else:
-            logger.error(
-                f"`role` config value {self.config.get('role')!r} invalid: should "
-                f"be one of {list(TempoRole)}."
-            )
+        # add Tempo worker custom blocking conditions
+        if self.worker.roles and self.worker.roles[0] not in self._valid_roles:
             e.add_status(
                 BlockedStatus(f"Invalid `role` config value: {self.config.get('role')!r}.")
             )
 
     @property
     def tempo_endpoint(self) -> Optional[str]:
         """Tempo endpoint for charm tracing."""
-        return self.tempo_cluster.get_tracing_endpoint("otlp_http")
+        if endpoints := self.worker.cluster.get_tracing_receivers():
+            return endpoints.get("otlp_http", None)
 
     @property
     def ca_cert_path(self) -> Optional[str]:
         """CA certificate path for tls tracing."""
-        return TEMPO_CA_FILE if Path(TEMPO_CA_FILE).exists() else None
+        return CLIENT_CA_FILE
+
+    def pebble_layer(self, worker: Worker) -> Layer:
+        """Return a dictionary representing a Pebble layer.
+
+        Caller is responsible for checking whether the tempo role is valid before
```

Comment:
This makes it sound like we'll raise an exception if not, while all that we get is a broken deployment. 
Rephrase, perhaps with:

`This method assumes that self.worker.roles is valid.`